plugin/src/main/groovy/com/novoda/staticanalysis/internal/VariantAware.groovy

```diff
@@ -0,0 +1,43 @@
+package com.novoda.staticanalysis.internal
+
+import org.gradle.api.DomainObjectSet
+import org.gradle.api.NamedDomainObjectSet
+
+trait VariantAware {
+
+    private boolean hasFilter = false
+    Closure<Boolean> includeVariantsFilter = { true }
```

Comment:
this can be just `Closure<Boolean> getIncludeVariantsFilter()` that we'll leave to the configurators to implement.

Reply:
`setter` is also used in subclasses. I can of course do this but the beauty of a trait is to encapsulate the variant behavior all here.
If I implement `getIncludeVariantsFilter` in the subclasses, then I would need a property there.